### ISSUE TITLE
Two issues created since using user.js. I'm not very technical, and would appreciate help.

### ISSUE THREAD

**Issue by lovfi** [arkenfox/user.js#1621]
I've just finished going through the wiki, and creating my overrides file. However, I am experiencing two problems with my browser.

1. **Spaces on either side of the webpage.** I think this has to do with that RFP setting, however, when not enabling it the issue occurs, and when I added a line to disable it, it continues to occur ([(img)](https://i.imgur.com/H3QYz3D.jpg)
2. Anything searched in search bar is not being searched with my search engine, but is going to a site (e.g. searching "frog" visits the site "https://frog/"). I believe I made the correct overrides for the search related preferences, but I am having this issue, so it seems I've made a mistake.

I'm terribly sorry for not being good at this. I'm trying my best to get into privacy, but without any background knowledge of much of this, it has been a bit of an uphill swim. I've spent a few hours going through all the documentation, and a good 20 minutes trying to fix these two problems, but I haven't made much progress on the latter. If anyone could help me out, that would be great.

My overrides file is:
```js
// allows use of searchbar:
user_pref("keyword.enabled", true);
user_pref("browser.search.suggest.enabled", true);
user_pref("browser.urlbar.suggest.searches", true);
// antivirus fix 
user_pref("security.cert_pinning.enforcement_level", 1);
// fix auth and third party content 
user_pref("network.http.referer.XOriginPolicy", 0);
// allow streaming services
user_pref("media.eme.enabled", true);
// keep history on close 
user_pref("privacy.clearOnShutdown.history", false);   // [DEFAULT: true]
// webGL
user_pref("webgl.disabled", false);
// https over DNS 
user_pref("network.trr.uri", "https://mozilla.cloudflare-dns.com/dns-query");
user_pref("network.trr.mode", 2);
// disables safe browsing
user_pref("browser.safebrowsing.malware.enabled", false);
user_pref("browser.safebrowsing.phishing.enabled", false);
user_pref("browser.safebrowsing.downloads.enabled", false);
// Leave IPv6 enabled
user_pref("network.dns.disableIPv6", false);
// Disk caching, which might improve performance if enabled.
user_pref("browser.cache.disk.enable", false);
//Enable favicons, the icons in bookmarks
user_pref("browser.shell.shortcutFavicons", true);
// start on home page
user_pref("browser.startup.page", 1);
/* 0103: set HOME+NEWWINDOW page
 * about:home=Firefox Home (default, see 0105), custom URL, about:blank
 * [SETTING] Home>New Windows and Tabs>Homepage and new windows ***/
user_pref("browser.startup.homepage", "about:Firefox Home");
/* 0104: set NEWTAB page
 * true=Firefox Home (default, see 0105), false=blank page
 * [SETTING] Home>New Windows and Tabs>New tabs ***/
user_pref("browser.newtabpage.enabled", true);
/*** [SECTION 1700]: CONTAINERS ---REMOVING FOR SIMPLICITY ***/
user_pref("_user.js.parrot", "1700 syntax error: the parrot's bit the dust!");
/* 1701: enable Container Tabs and its UI setting [FF50+]
 * [SETTING] General>Tabs>Enable Container Tabs
 * https://wiki.mozilla.org/Security/Contextual_Identity_Project/Containers ***/
user_pref("privacy.userContext.enabled", false);
user_pref("privacy.userContext.ui.enabled", false);
/* 1702: set behavior on "+ Tab" button to display container menu on left click [FF74+]
 * [NOTE] The menu is always shown on long press and right click
 * [SETTING] General>Tabs>Enable Container Tabs>Settings>Select a container for each new tab ***/
   // user_pref("privacy.userContext.newTabContainerOnLeftClick.enabled", true);

/*** DISABLING RFP ***/
user_pref("privacy.resistFingerprinting", false); 
user_pref("privacy.resistFingerprinting.letterboxing", false); 
user_pref("webgl.disabled", false);
user_pref("privacy.resistFingerprinting.letterboxing.dimensions", false);
```

**Comment by Thorin-Oakenpants** [comment]
put your overrides in a **SEPARATE FILE** (`user-overrides.js`) and **THEN** run the updater. The updater updates the users.js to match here, and then **ADDS** your overrides to the end
- see https://github.com/arkenfox/user.js/wiki/3.4-Apply-&-Update-&-Maintain
- it looks like you are not applying any overrides

yes
- `privacy.resistFingerprinting.letterboxing` is the letterboxing: margins around the web content
- `keyword.enabled` - enabled you to type `frog` and hit enter and go to your search engine
- as I said, sounds like you are not applying ANY overrides

---

This is not a full check, just pointing out some obvious things

remove this from your overrides: first it is not a boolean (true, false) so it has no effect (type mismatch) and secondly, it is inactive (commented out) in the user.js so there is no need for you to do anything with it
```js
user_pref("privacy.resistFingerprinting.letterboxing.dimensions", false);
```

fix this
```js
user_pref("browser.startup.homepage", "about:home"); // it is not `about:Firefox Home`
```

add this to the START of your `user-overrides.js`
```js
user_pref("_user.js.parrot", "custom overrides: starting");
```

add this to the END of your `user-overrides.js`
```js
user_pref("_user.js.parrot", "custom overrides: finished .. success");
```

then you can check in about:config and type in `parrot` and it should say the pref value - i.e success. You can also check your console right after starting Firefox (control-J) and check for any errors to do with the user.js

**Comment by Thorin-Oakenpants** [comment]
also: if you want IPv6 enabled and you want to enabled the disk cache, then the values should be `true`

pay attention to what you add

```js
// Leave IPv6 enabled
user_pref("network.dns.disableIPv6", false);
// Disk caching, which might improve performance if enabled.
user_pref("browser.cache.disk.enable", false);
```

**Comment by lovfi** [comment]
I think you're right about no changes being applied. However, I was already doing this with a separate user-overrides.js file, and I've been running the updater.bat file just fine without any errors. I've tried a few more times after making the changes to the overrides you suggested (thanks for those btw!), but I'm still not seeing anything being applied.

Checking parrot in about:config just says "SUCCESS: No no he's not dead, he's, he's restin'!"

These are the errors I got after starting firefox (and going to github first, probably should've taken the ss before that-) : [img](https://i.imgur.com/fBU04Qd.jpg)

I've followed your instructions and the wikis exactly as I read them, but I'm still not getting anywhere. Do you have any ideas of what could be wrong still?

**Comment by Thorin-Oakenpants** [comment]
Are you putting the files in the right directory?

Open `about:support` and the first list is `Application Basics` .. and the 10th item in it is your `Profile folder` which lists the dir and also has an `open folder` button

**Comment by lovfi** [comment]
I'm just adding one last comment here along with closing this issue as I believe things are now resolved.

As for what I did to fix the issue I was having--I honestly have no clue. I had been using a secondary browser in the meantime until I had this sorted out, and after two days of doing everything I could and giving it a rest, I decided to try once more. I opened firefox as usual, but surprisingly, without doing anything since last opening the browser, my overrides had been applied and the user.js was all working as intended. 

I really don't know what changed, but it's working just fine now. I was always using the correct profile, and I had restarted the browser and my computer multiple times while I was experiencing the issue, but none of that worked then, so I'm not sure what happened in between so that it's working fine now. I just ran the updater once more as v109 was just released, and it had no issue, so I think I can safely say I won't be having the issue anymore.

Thanks for the help Thorin-Oakenpants! I am happy to now be using arkenfox user.js 😄
